src/components/standalone/chart/StandaloneChart.tsx

```diff
@@ -0,0 +1,54 @@
+import { ReactChildren, ReactChild } from "react";
+import PublishButton from "../PublishButton";
+import "./standalone-chart.css";
+import { useContext } from "react";
+import ChartContext from "../../../context/ChartContext";
+import StandaloneContext from "../../../context/StandaloneContext";
+import CSVUploader from "../csv-uploader/CSVUploader";
+import SparqlInput from "../sparql-input/SparqlInput";
+
+interface Props {
+  children: ReactChild | ReactChildren;
+}
+
+const ChartPreviewStandalone = ({ children }: Props) => {
+  const { chartDefinition, setChartDefinition, chartProperties }: any =
+    useContext(ChartContext);
+  const { isFullScreen, setIsFullScreen } = useContext(StandaloneContext);
+
+  const emptyDataState = Object.keys(chartDefinition).length === 0;
+  const isAMap = chartProperties.chartTypes.chartType === "Map";
+
+  if (emptyDataState) {
+    return isAMap ? <SparqlInput /> : <CSVUploader />;
+  }
+
+  const onFullScreenClick = () => {
+    setIsFullScreen(!isFullScreen);
+    const { layout } = chartDefinition;
+    // increment datarevision to force plotly to update the chart width.
+    layout.datarevision++;
+    setChartDefinition({ ...chartDefinition, layout: { ...layout } });
```

Comment:
`layout.datarevision++` still mutates the underlying `layout` value.

This should be .... `layout: { ...layout, datarevision: layout.datarevision + 1 }`

Reply:
Tweaked - thanks